api/integrations/slack/views.py

```diff
@@ -0,0 +1,91 @@
+import uuid
+
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.shortcuts import redirect
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from slack_sdk.oauth import AuthorizeUrlGenerator
+
+from integrations.common.views import IntegrationCommonViewSet
+from integrations.slack.models import SlackConfiguration, SlackEnvironment
+from integrations.slack.serializers import (
+    SlackChannelListSerializer,
+    SlackEnvironmentSerializer,
+)
+from integrations.slack.slack import get_bot_token, get_channels_data
+
+from .exceptions import InValidStateError
+
+
+class SlackConfigurationViewSet(IntegrationCommonViewSet):
+    serializer_class = SlackEnvironmentSerializer
+    pagination_class = None  # set here to ensure documentation is correct
+    model_class = SlackEnvironment
+
+    @action(
+        detail=False,
+        methods=["GET"],
+        url_path="channels",
+        serializer_class=SlackChannelListSerializer,
+    )
+    def get_channels(self, request, *args, **kwargs):
+        env = self.get_environment_from_request()
+
+        try:
+            config = SlackConfiguration.objects.get(project=env.project)
+        except ObjectDoesNotExist:
+            return Response(
+                "Slack api token not found. Please generate the token using oauth",
+                status.HTTP_400_BAD_REQUEST,
+            )
+        slack_token = config.api_token
+        serializer = self.get_serializer(data=get_channels_data(slack_token), many=True)
+        serializer.is_valid()
+        return Response(serializer.data)
+
+    def _get_slack_callback_url(self, **kwargs):
```

Comment:
This should live at the bottom of the class rather than sitting between 2 action routes

Reply:
Seems like we're only using it in one place too - I'm not sure it _needs_ extracting to a method like this does it? Wouldn't it be just as obvious what's happening if we do `slack_callback_url = self.reverse_action(...)` ?

Reply:
Nope we are using at two places, and we are passing it directly to the function call(save a line ;) )